develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md

```diff
@@ -0,0 +1,96 @@
+---
+uid: NuGet_Naming_Conventions
+---
+
+# NuGet naming conventions
+
+> [!IMPORTANT]
+> This section might include some information that is only applicable to Skyline employees and/or links that are only accessible to Skyline employees.
+
+Below rules are mainly focused for NuGets that are published to nuget.org. It is adviced to use the same rules for internal NuGets as well as that would make it easy transferable to nuget.org when necessary.
+
+## Mandatory rules
+
+- NuGet must have a name that starts with 'Skyline.DataMiner'.
+- NuGet must include a license file when published to nuget.org
+- NuGet must be signed otherwise this will be blocked by nuget.org
+    - When creating the NuGet via [SLC SE Repository Manager](xref:TOOSLCSERepositoryManager), the pipeline will take care of the signing and publishing.
+- [Mandatory metadata](#mandatory-metadata)
+
+<br/>
+
+- If there is a dependency on a DataMiner DLL (from the Skyline DataMiner/Files folder), use the corresponding NuGet (Skyline.DataMiner.Files.XXX).
+    - If it is not there, contact the [Data-Acquisition](mailto:support.data-acquisition@skyline.be) domain. They can then include this DLL in the automatic NuGet creation from DataMiner versions.
+- If there is a dependency on multiple DataMiner DLLs that are default included in a QAction/Exe, then make use of the DevPacks (Skyline.DataMiner.Dev.XXX)
+
+<br/>
+
+- Use the syntax: 'Skyline.DataMiner.\[Categories\].\[Name\]'. Optionally at the end you could have '.Protocol', '.Automation', '.Common', ... to specify on where it can be used (usage of SLManagedScripting, SLManagedAutomation, ...)
+    - Skyline.DataMiner.\[Categories\].\[Name\].Common (Shared code that can run either in Script or Protocol. Communicates with DataMiner via SLNet or [DataMinerSystem](https://www.nuget.org/packages?q=Skyline.DataMiner.Core.DataMinerSystem))
+    - Skyline.DataMiner.\[Categories\].\[Name\].Automation (Acts as entrypoint for Automation (methods for IEngine) or has a dependency on classes from SLManagedAutomation)
+    - Skyline.DataMiner.\[Categories\].\[Name\].Protocol (Acts as entrypoint for Protocol (methods for SLProtocol) or has a dependency on classes from SLManagedScripting)
+
+    - Another situation is the module that it has to do with. An example of this is the CI/CD packages:
+        - Skyline.DataMiner.CICD.DMApp.Common (common code for creating a package)
+        - Skyline.DataMiner.CICD.DMApp.Automation (specific code to convert an Automation solution to dmapp package)
+        - Skyline.DataMiner.CICD.DMApp.XXX (specific code to convert a XXX solution/repository to dmapp package)
+
+## Mandatory metadata
+
+```xml
+<PropertyGroup>
+    ...
+    <Authors>SkylineCommunications</Authors>
+    <Company>Skyline Communications</Company>
+    <Descrption>{Short description about the NuGet}</Description>
```

Comment:
I would add here that if there is a specific line of code as an "entry point" it should be added here. This makes it much more easy to start using any library.  {Code Entry Point: line of code that gets a developer started with your library}
Example:
Library facilitating communication between DataMiner elements and automationscripts by providing a Command Pattern Framework. Code Entry Point: IInterAppCall command = InterAppCallFactory.CreateNew();

Reply:
Actually: I also added [Minimum Required DataMiner: 10.1.0.6]  to the start of the description for the Class Library things. So it's clear.

<Description>{[Minimum Required DataMiner: A.B.C.D} {Short description about the NuGet} {Code Entry Point: line of code that gets a developer started with your library}</Description>

The description is the first thing a user generally see's in the NuGetManager.
Also you misspelled the first opening tag: Descrption   should be Description

Reply:
Not sure if the entry point should be part of the description. I feel it belongs more to the readme file.

Reply:
@janstaelensskyline : I'm also in favour of using SDK style projects by default